pricegraph/src/graph.rs

```diff
@@ -0,0 +1,5 @@
+//! This module contains graph related data types and algorithms that are either
+//! not included in the `petgraph` crate or are modified versions of the crate's
+//! implementation.
+
+pub mod bellman_ford;
```

Comment:
I understand the Bellmann-Ford is very similar Dijkstra for the same problem but can handle negative values in edges weights. I also see a lot of reference to negative cycles which can only be the result of having negative edges. Does this mean our order graph has negative edges? I can't think of any edge weights in an orderbook that would be negative.

If there are no negative edges, it is my understanding that Dijkstra's algorithm would be the superior choice.

Reply:
Negative edges happen because we use the logarithm of the price which can be negative. Dijkstra won't work here, as multiplication with factors < 1 are equivalent to addition with negative edges which does not work with Dijkstra.

It would be nice if we could as in general Dijkstra is faster for dense graphs.